docs/nix-develop.md

```diff
@@ -0,0 +1,203 @@
+# Using Nix Develop Environment
+
+## Install nix from shell
+
+    sh <(curl -L https://nixos.org/nix/install) --daemon
+
+then `exec bash -l` reload your bash to load the nix-env
+
+## Install flake feature of nix
+
+    nix-env -iA nixpkgs.nixUnstable
+    echo "experimental-features = nix-command flakes" | sudo tee -a /etc/nix/nix.conf
+    sudo pkill nix-daemon
+
+ NOTE: `reload bash` and check your nix-version first
+
+    nix --version
+    nix (Nix) 2.4pre20210326_dd77f71
+  
+## Clone Emacs-ng by nix-shell and enable emacs cachix 
+
+```bash
+nix-env -iA cachix -f https://cachix.org/api/v1/install
+exec bash -l
+cachix use emacsng # make sure you have saw the output like: Configured https://emacsng.cachix.org binary cache in /home/test/.config/nix/nix.conf
+nix-shell -p git --command "git clone https://github.com/emacs-ng/emacs-ng.git && cd emacs-ng && nix-shell"
+```
+
+## Setting up Rust develop environment
```

Comment:
Is this all relevant in order to install emacs-ng ?

Reply:
yes, it's necessary. I should explain the information to users on how to modify the build process by themselves. such as `configFlags` `patching the emacs-ng` `rust-nightly version`